src/repositories/statusReasonRepository.ts

```diff
@@ -0,0 +1,18 @@
+import { ProjectStatusReason } from '../entities/projectStatusReason';
+
+export const findAllStatusReasons = async (): Promise<
+  ProjectStatusReason[]
+> => {
+  return ProjectStatusReason.createQueryBuilder('project_status_reason')
+    .leftJoinAndSelect('project_status_reason.status', 'status')
+    .getMany();
+};
+
+export const findStatusReasonsByStatusId = async (
+  statusId: number,
+): Promise<ProjectStatusReason[]> => {
+  return ProjectStatusReason.createQueryBuilder('project_status_reason')
+    .leftJoinAndSelect('project_status_reason.status', 'status')
+    .where(`"statusId" = ${statusId}`)
```

Comment:
Small change here, now that we are refactoring. Interporlation like this prone to SQL injection, maybe not in this case. But the best practices for ORMs and queries, is letting the ORM interpolate and escape the characters, preventing sql injection.
We are switching any interpolation to this format:
```javascript
.where('project_status_reason."statusId" = :statusId', { statusId: statusId })
```

Reply:
Thanks @CarlosQ96  for your review, I didn't know it could be vulnerable, good point.
Done.